pennylane/debugging/operation_checker.py

```diff
@@ -0,0 +1,833 @@
+                # Check number of sets of parameters to match the number of operations
+                assert len(parameters) == len(op)
+            if wires is None or isinstance(wires, qml.wires.Wires):
+                # Broadcast the wires
+                wires = [wires] * len(op)
+            else:
+                # Check number of sets of wires to match the number of operations
+                assert len(wires) == len(op)
+
+        else:
+            op = [op]
+            parameters = [parameters]
+            wires = [wires]
+
+        self.seed = seed
+        # Initialize result status for all operations
+        self.results = {op_: "pass" for op_ in op}
+        self.output = {}
+        for op_, parameters_, wires_ in zip(op, parameters, wires):
+            # Temporary storage per operation
+            self.tmp = {
+                "printed_header": False,  # Header for this op has not been printed yet
+                "op": op_,  # Currently investigated operation
+                "res": max(verbosity_levels.values()),  # Current result status
+                "name": op_.name if isinstance(op_, qml.operation.Operation) else op_.__name__,
+            }
+            self.output[op_] = ""
+
+            try:
+                self.check_single_operation(op_, parameters_, wires_)
+            except CheckerError as e:
+                pass
+
+            # Store the result for this operation outside of tmp and print summary per op
+            self.results[op_] = levels_verbosity[self.tmp["res"]]
+            if self.results[op_] == "pass":
+                self.print_(
+                    f"No problems have been found with the operation {self.tmp['name']}.\n", "pass"
+                )
+
+        return self.results, self.output
+
+    def print_(self, string, level=None):
+        """Print a string if the verbosity level allows it, color it if applicable,
+        and increment the result status for the currently checked operation if necessary.
+
+        Args:
+            string (str): String to be printed
+            level (str): One of the verbosity levels (see class documentation)
+                If the level is in the levels that are printed, print the string to console
+                and store it in ``self.output``.
+
+        Returns:
+
+        A header is printed whenever a ``print_`` statement is executed first for
+        a given operation (and the verbosity levels actually allow for an output).
+        """
+
+        self.tmp["res"] = min(self.tmp["res"], verbosity_levels[level])
+
+        # Colorize the string
+        if self.print_color and level is not None:
+            string = f"\033[{_colors[level]}m{string}\033[0m"
+
+        # Errors are always printed
+        if level == "error" or level in self._verbosity:
+            if not self.tmp["printed_header"]:
+                header = f"Checking operation {self.tmp['name']} for consistency.\n" + "= " * 40
+                print(header)
```

Comment:
@dwierichs I recommend _not_ hardcoding `print()` anywhere here, but instead allowing users to configure what their 'print function' is. E.g., maybe they want to use a logger.

Reply:
I went for both now. This makes it possible to retrieve a log of all checks :)